qt/python/mantidqtinterfaces/mantidqtinterfaces/Muon/GUI/Common/pairing_table_widget/pairing_table_widget_presenter.py

```diff
@@ -49,39 +40,81 @@ def notify_data_changed(self):
         self._dataChangedNotifier()
 
     def disable_editing(self):
-        self._view.disable_editing()
+        self.disable_updates()
+        self._view.set_is_disabled(True)
+        self._view.disable_all_buttons()
+        self._disable_all_table_items()
+        self.enable_updates()
 
     def enable_editing(self):
-        self._view.enable_editing()
+        self.disable_updates()
+        self._view.set_is_disabled(False)
+        self._view.enable_all_buttons()
+        self._enable_all_table_items()
+        self.enable_updates()
+
+    def enable_updates(self):
+        """Allow update signals to be sent."""
+        self._view.set_is_updating(False)
+
+    def disable_updates(self):
+        """Prevent update signals being sent."""
+        self._view.set_is_updating(True)
+
+    def num_rows(self):
+        return self._view.get_pairing_table.rowCount()
+
+    def num_cols(self):
+        return self._view.get_pairing_table.columnCount()
+
+    def remove_pair_by_index(self, index):
+        self._view.get_pairing_table.removeRow(index)
+
+    def remove_last_row(self):
+        last_row = self._view.get_pairing_table.rowCount() - 1
+        if last_row >= 0:
+            self._view.get_pairing_table.removeRow(last_row)
+
+    def clear(self):
+        # Go backwards to preserve indices
+        for row in reversed(range(self.num_rows())):
+            self._view.get_pairing_table.removeRow(row)
+
+    def get_selected_row_indices(self):
+        return list(set(index.row() for index in self._view.get_pairing_table.selectedIndexes()))
+
+    def get_selected_pair_names_and_indexes(self):
+        indexes = self.get_selected_row_indices()
+        return [[str(self._view.get_pairing_table.item(i, 0).text()), i] for i in indexes]
 
     def handle_guess_alpha_clicked(self, row):
-        table_row = self._view.get_table_contents()[row]
+        table_row = self.get_table_contents()[row]
         pair_name = table_row[0]
         group1 = table_row[2]
         group2 = table_row[3]
         self.guessAlphaNotifier.notify_subscribers([pair_name, group1, group2])
 
     def handle_data_change(self, row, col):
-        table = self._view.get_table_contents()
+        table = self.get_table_contents()
         changed_item = self._view.get_table_item(row, col)
-        changed_item_text = self._view.get_table_item_text(row, col)
-        pair_name = self._view.get_table_item_text(row, 0)
+        changed_item_text = self.get_table_item_text(row, col)
+        pair_name = self.get_table_item_text(row, 0)
         update_model = True
-        if pair_columns[col] == "pair_name" and not self.validate_pair_name(changed_item_text):
+        if get_pair_columns()[col] == "pair_name" and not self.validate_pair_name(changed_item_text):
             update_model = False
```

Comment:
I was thinking that the return statement might not be ideal because line 127, which is the final condition that checks if col != 1, would not be triggered if a return statement is used. What are your thoughts?

if col != 1:
            self.update_view_from_model()
            self.notify_data_changed()

Reply:
Ah, good catch, that part was hidden by github so I hadn't noticed! Thanks